### PR TITLE
Continue on errror

### PR DESCRIPTION
**Pull request by MrSeccubus**
As a user I work with email addresses that I get from an external party (in this case abuse email addreses from ipinfo.io)

Sometimes these email addresses contain errors that cause the smtp server to report them back to me and fail. I would like it if the software automatically continues/resumes with the next record instead of having to do that manually.

Also I would like to have a log file (in csv format) so that I know which email were not sent.

### CODE REVIEW

**Comment by MrSeccubus** [comment]
I see that the code fails to run on Python 2, but should we still deal with Python 2 given that it is an unsupported language?

Also I don't know how to address the codecov problems.

**Comment by MrSeccubus** [comment]
This pull request should also address issue #96

**Comment by MrSeccubus** [comment]
@awdeorio the unit test that is failing is for a python version that is out of support. DO you really want me to fix that specific error?
Otherwise is there anything else that is holding up this PR?

**Comment by awdeorio** [comment]
> @awdeorio the unit test that is failing is for a python version that is out of support. DO you really want me to fix that specific error?
> Otherwise is there anything else that is holding up this PR?

Thanks for your patience.  I know I've been slow on this and #114. I'll take a closer look at this after finishing up #114 and getting it merged.

File: mailmerge/__main__.py
Comment (awdeorio) [inline mailmerge/__main__.py:14]: I would lean towards using Python's standard library [logging](https://docs.python.org/3/howto/logging.html) mechanism over building a custom logger.

File: mailmerge/__main__.py
Comment (awdeorio) [inline mailmerge/__main__.py:87]: I think that creating a log by default is reasonable.  Maybe we can just call it `mailmerge.log`?  That would simplify the command line interface.

File: mailmerge/__main__.py
Comment (awdeorio) [inline mailmerge/__main__.py:129]: I think this logic can be simplified after merging in the latest upstream develop.  That includes a try-except block in the message send part.

File: tests/test_main_log_and_continue.py
Comment (awdeorio) [inline tests/test_main_log_and_continue.py:1]: More ideas for tests:
- Log file already exists

**Review by awdeorio — CHANGES_REQUESTED** [review]
Sorry for the slow review.  Thanks for your contributions!  I left some comments.

**Comment by MrSeccubus** [comment]
Continueing this work in PR #124

**Pull request closed without merging**
